### ISSUE TITLE
props spread in component does not work as expected

### ISSUE THREAD

**Issue by atfzl** [ryansolid/solid#75]
I have reproduced this issue in https://codesandbox.io/s/solid-counter-ubcwc

```js
function Counter(props) {
  return <div {...props} />
}

function Main() {
  return <Counter>{(state.counter)}</Counter>;
}
```
It gives a `TypeError: Cannot assign to read only property 'children' of object '#<HTMLDivElement>'`
We use this pattern a lot to passing all props to the inner component/element in React.

Right now the workaround is to explicitly pass children.
```js
function Counter(props) {
  return <div>{(props.children)}</div>;
}

function Main() {
  return <Counter>{(state.counter)}</Counter>;
}
```

**Comment by atfzl** [comment]
Also the spread parameters are not passed to the inner element/component even after removing children.

```js
function Counter(props) {
  const { children, ...rest } = props;

  /* prettier-ignore */
  return <div {...rest}>{(props.children)}</div>;
}

function Main() {
  return <Counter onClick={console.log}>{(state.counter)}</Counter>
}
```

Here the onClick is not being passed to the inner component.

**Comment by ryansolid** [comment]
Thanks for bringing this to my attention. The first issue should have been obvious to me in hindsight. I will check it out. The 2nd issue surprises me. The first I get since it's trying to set the children property on a DOM node. I can special case that pretty easily. I guess making it its own insert call on detecting children property. The 2nd issue should generally work, unless it is something specific to delegated event handlers. I should point out there is a dynamic version of spreads as well. I definitely need to get this working well to avoid manual destructuring since that would trigger dependency tracking.

**Comment by ryansolid** [comment]
Ok fixed in 0.10.7. I also needed to make the spread dynamic. See the modified version of your example here: https://codesandbox.io/s/solid-counter-spread-4vtvr.

**Comment by atfzl** [comment]
Thanks @ryansolid 

I am working on a real world project with solid before i pitch it at work place, so i'll let you know if there are any hiccups.
